src/game/server/entities/character.cpp

```diff
@@ -2841,12 +2875,10 @@ bool CCharacter::TakeDamage(vec2 Force, int Dmg, int From, int Weapon, int Mode)
 	// m_pPlayer only inflicts half damage on self
 	if(From == m_pPlayer->GetCID())
 	{
-		if(GetClass() == PLAYERCLASS_HERO || (GetClass() == PLAYERCLASS_SOLDIER && m_ActiveWeapon == WEAPON_GRENADE)
-										  || (GetClass() == PLAYERCLASS_SCIENTIST && Mode == TAKEDAMAGEMODE_NOINFECTION)
-										  || (GetClass() == PLAYERCLASS_LOOPER && m_ActiveWeapon == WEAPON_GRENADE))
-			return false; // no self harm
-		else
+		if (Mode == TAKEDAMAGEMODE_SELFHARM)
```

Comment:
how is selfharm related to turrets? It should be changed/fixed in a separate pr

Reply:
it is required, otherwise the turrets will harm the owner of the turrets if they hit him

Reply:
No further splitting from my side. Do it yourself or ask yavl to dump this feature.